plugin/evm/vm.go

```diff
@@ -128,8 +129,26 @@ const (
 	bytesToIDCacheSize  = 5 * units.MiB
 
 	targetAtomicTxsSize = 40 * units.KiB
+
+	// p2p app protocols
+	ethTxGossipProtocol    = 0x0
+	atomicTxGossipProtocol = 0x1
+
+	// gossip constants
+	txGossipMaxResponseSize        = 20 * units.KiB
```

Comment:
What happens if there is a single transaction that is this large (like a contract deploy)? I'm assuming it sends the first item, even if it exceeds?

Reply:
Nonetheless, I think we may want to at least set this larger than the max contract deploy size.

Reply:
In this case we canter gossip it. I based this by seeing that most txs are between 300B to 1KB... but theoretically I think our max message size is a MB but that sounds too high. Ideally this is the p100 of our tx size but I don't know if we define one.

Reply:
Seems pretty easy to block the mempool then by issuing 1 contract deploy larger than 20 KiB with the highest fee?

Reply:
Unless we want to avoid gossiping things above a certain size?

Reply:
Replaced w/ target gossip size